### PR TITLE
Use env variable in healthcheck

### PR DESCRIPTION
**Pull request by pbek**
It would be better to use the env variable here so the secret env variables can go to a `env_file`.

### CODE REVIEW

**Comment by pbek** [comment]
(I run that in production)

**Review by Schrolli91 — APPROVED** [review]
Absolute with you - LGTM

File: docs/docker-compose.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ services:
     command: --default-storage-engine innodb
     restart: unless-stopped
     healthcheck:
-      test: mysqladmin -pchangemeplease ping -h localhost
+      test: mysqladmin -p$$MYSQL_ROOT_PASSWORD ping -h localhost
```
Comment (kevinpapst) [inline docs/docker-compose.md:23]: out of curiosity: why are there two $ ?

**Comment by tobybatch** [comment]
Agreed, but... I'd like it to work out of the box so how about `${ROOTPASS:-changemeplease}`?

Also that will need changing everywhere, so in compose/docker-compose.mysql.yml, docker-compose.yml, docs/dev-instance.md, and docs/docker-compose.md

File: docs/docker-compose.md (hunk 1 above)
Comment (tobybatch) [reply docs/docker-compose.md:23]: There shouldn't be.

Comment (pbek) [reply docs/docker-compose.md:23]: Docker Compose can't access the env variables at that time in there. I had to try multiple ways. This way was the way that finally worked (stackoverflow helped a lot 😁).

**Comment by pbek** [comment]
>  I'd like it to work out of the box

It works out of the box, because the env var is set in your example. Or what where you referring to?

**Comment by tobybatch** [comment]
Fair enough but someone will raise an issue about it not working when they don't bother to read the example properly. By setting a default, it will always work.

**Comment by pbek** [comment]
> Fair enough but someone will raise an issue about it not working when they don't bother to read the example properly.

Sorry, I can't follow. In what scenario wouldn't it work?

> By setting a default, it will always work.

I doubt it would work since the env variable can't be accessed directly and I don't know if the `$$` works with your example (or did you test it, then it would be great).

File: docs/docker-compose.md (hunk 1 above)
Comment (kevinpapst) [reply docs/docker-compose.md:23]: Is that the part of the relevant docs: https://docs.docker.com/compose/compose-file/compose-file-v2/#variable-substitution ?

> You can use a $$ (double-dollar sign) when your configuration needs a literal dollar sign. This also prevents Compose from interpolating a value, so a $$ allows you to refer to environment variables that you don’t want processed by Compose.

Comment (pbek) [reply docs/docker-compose.md:23]: That looks right! Thank you!

**Comment by tobybatch** [comment]
And the scales fall from my eyes...

Sorry, I get it now.

**Comment by pbek** [comment]
I almost thought I was missing something. 😅

**Pull request merged**